pkg/manager/member/pd_failover.go

```diff
@@ -121,7 +127,20 @@ func (pf *pdFailover) tryToMarkAPeerAsFailure(tc *v1alpha1.TidbCluster) error {
 	ns := tc.GetNamespace()
 	tcName := tc.GetName()
 
+	pdsts, err := pf.setLister.StatefulSets(ns).Get(controller.PDMemberName(tcName))
+	if err != nil {
+		return err
+	}
+	podNames := sets.String{}
+	for _, ordinal := range helper.GetPodOrdinals(*pdsts.Spec.Replicas, pdsts).List() {
+		podNames.Insert(util.GetPodName(tc, v1alpha1.PDMemberType, ordinal))
+	}
+
 	for podName, pdMember := range tc.Status.PD.Members {
+		if !podNames.Has(podName) {
```

Comment:
Should we keep consistent with tikv handling [here](https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/tikv_member_manager.go#L605)?

Reply:
updated.